sdk-setup/src/rpmvalidation

```diff
@@ -0,0 +1,206 @@
+    read -d '' -r python <<EOF
+import sys, configparser;
+config = configparser.RawConfigParser();
+config.read(sys.argv[2:]);
+sec, val = sys.argv[1].split(".");
+print(config.getboolean(sec,val));
+EOF
+    sb2 -t "$OPT_TARGET" python3 -c "$python" "RpmValidationSuite.$key" "${ini_files[@]}"
+}
+ 
+list_target() {
+    local target=$1
+    local display_target=$2
+    local f=
+    for f in "${SUITES[@]}"; do
+        local essential=$(suite_boolean "$target" "$f" Essential)
+        if [[ "$essential" == "True" ]]; then
+            essential="Essential"
+        else
+            essential="Optional"
+        fi
+        local website=$(suite_value "$target" "$f" Website)
+        local name=$(suite_value "$target" "$f" Name)
+        echo "$display_target" "$f" "$essential" "$website" "$name"
+    done
+}
+
+list_all() {
+    for t in $(sdk-manage target list); do
+        populate_suites $t
+        list_target $t $t
+    done
+}
+
+run_suite() {
+    local suite=$1
+    suite_exec=$(suite_value "$OPT_TARGET" "$suite" Exec)
+    suite_cmd=${suite_exec//%f/$RPM_FILE}
+
+    sb2 -t "$OPT_TARGET" "$suite_cmd"
+}
+
+run_all_suites() {
+    echo "Using validation suites: ${SUITES[@]}"
+    local f=
+    for f in "${SUITES[@]}"; do
+        run_suite $f
+    done
+}
+
+set_defaults()
+{
+    OPT_TARGET=
+    RPM_FILE=
+    OPT_USAGE=
+    OPT_LIST=
+}
+
+parse_options() {
+    if [[ $0 == *.sh ]]; then
+        echo "Warning: The name '$0' is deprecated. Use '${0%.sh}' instead" >&2
+    fi
+
+    while [[ $1 ]]; do
+        case $1 in
+            "-h" | "--help")
+                OPT_USAGE=1
+                return
+                ;;
+            "-t" | "--target")
+                shift
+                OPT_TARGET=$1
+                ;;
+            "-l" | "--list-suites")
+                OPT_LIST=1
+                ;;
+            *)
+                RPM_FILE=$1
+                ;;
+        esac
+        shift
+    done
+
+    if [[ ! $OPT_LIST ]]; then
+        if [[ ! $RPM_FILE ]]; then
+            OPT_USAGE=1
+        fi
+    fi
+}
+
+main() {
+    set_defaults
+
+    parse_options "$@" || return
+
+    if [[ $OPT_USAGE ]]; then
+        usage
+        return
+    fi
+
+    populate_suites $OPT_TARGET
+
+    if [[ $OPT_TARGET ]]; then
+        if [[ ${#SUITES[@]} -lt 1 ]]; then
+            echo "Target"
+            echo "======"
+            echo "ERROR [$OPT_TARGET] Target does not contain validation suites. Use a more recent target."
+            echo "FAILED"
+            return
+        fi
+    fi
+
+    if [[ $OPT_LIST ]]; then
+        if [[ $OPT_TARGET ]]; then
+            list_target $OPT_TARGET
+        else
+            list_all
+        fi
+        return
+    fi
+
+    if [[ ! $OPT_TARGET ]]; then
+        echo "Target"
+        echo "======"
+        echo "ERROR [] Target not specified"
+        echo "FAILED"
+        return
+    fi
+
+    echo "Running validation suites on target $OPT_TARGET"
+    run_all_suites
```

Comment:
It should be `run_essential_suites` instead. And we should also add the `--suites suite1[,suite2]...` switch to allow overriding this.

Reply:
The task specifically was to execute ALL suites.

Reply:
I wouldn't spent too much time on getting an ideal task granularity and try to finish it finally - it's a trivial difference on about 5 lines of code if I omit help text :)